packages/tupaia-web/vite.config.ts

```diff
@@ -40,6 +41,19 @@ export default defineConfig(({ command }) => {
         ...baseConfig.define,
         global: {},
       },
+      resolve: {
```

Comment:
I'm not sure how maintainable this is. We're basically re-specifying all dependencies of a frontend here?

The nice thing about the old way was that we don't have to do that, it reads the dependencies from package.json and generates the list of dirs to watch like that.

Reply:
This is just for hot reload so it's probably not a fair comparison to the old build tooling since hot reload didn't work with ui-components, it was just live reload and it was not very reliable. It's only for other frontend internal dependencies that are in the mono-repo so it will only ever be these 3 dependencies so I think it is pretty minor. It will be the exact same for the other front end apps when we move them to vite so we should probably make a shared vite config for the mono-repo in the same way that we do for babel and eslint and that should help ensure that the list of aliases in this file are easier to maintain.

Reply:
I see, sounds good, thanks for the info 🙏